ath79: change default config for dumb APs

StereoRocker opened the pull request:
Implements a uci-defaults script that, for boards which are dumb APs by
nature, changes the default network configuration as follows:

- Hostname is set to the serial number of the board, where possible.

This is primarily to avoid a DNS conflict on a network where the default
gateway already has the hostname "openwrt". A secondary bonus is that the
device would be easier to identify when looking at DHCP leases.

- Sets the LAN protocol to DHCP.

This is to stop the device taking the IP address 192.168.1.1 by default.
On many home networks this is the IP of the default gateway and, as such,
booting the device with this configuration while connected to the network
would cause an IP conflict.

- Removes references to WAN from default network config.

The boards targeted by this change have single ethernet ports, intended
for LAN. As such, a WAN interface is not expected to be used nor required.

Signed-off-by: Dominic Houghton <technpizza@gmail.com>

StereoRocker commented:
Sorry about the repeated force pushes. I had made some changes since I last tested the script, to try and make it better fit the format of other scripts that perform actions based on the board it runs on. That was the last that I'd need to do prior to review.

musashino205 commented:
```ucidef_set_interface_lan "eth0" "dhcp"``` is not enough?
See: https://github.com/openwrt/openwrt/blob/fb15cb4ce9559021d463b5cb3816d8a9eeb8f3f9/target/linux/ath79/generic/base-files/etc/board.d/02_network#L318-L320

StereoRocker commented:
> `ucidef_set_interface_lan "eth0" "dhcp"` is not enough?

That'd be enough to address setting LAN to DHCP. I still think there's value in setting default hostname for dumb AP boards to something unique, to avoid DNS conflicts. I'll change this script to do only that.

StereoRocker committed:
ath79: change default config for dumb APs

Implements a uci-defaults script that, for boards which are dumb APs by
nature, changes the default hostname to the serial number of the board.

Signed-off-by: Dominic Houghton <technpizza@gmail.com>

Hurricos reviewed (changes requested):
Would this change still leave `dnsmasq` serving DHCP and DNS on port 53? What about `odhcpd`?

I would suggest comparing to how the correct behavior is achieved in the `nas` `DEVICE_TYPE` (as opposed to `router` or `basic`, see e.g. `target/linux/oxnas/Makefile`) achieves the correct results.

mpratt14 commented:
for hostname

a similar thing was tried and reverted
22b8a6263d72bb1911c5841ef90ebdf452b53905

i also have a PR which refactors the function that was left behind in #9599

for dhcp

it's tricky because guides for first time use and failsafe mode all say the device will have a static address. Also, it should be possible to configure any board without needing another device to serve it an address. Otherwise in an imaginary scenario where the user just has the board and a laptop for some reason, they would have to set up DHCP server or open it for serial access.

for the WAN thing

sure why not, but would it be easy to set up WAN anyway? you never know when someone wants to use their dumb AP as something else
